Pepperdash Core/Pepperdash Core/Comm/GenericSecureTcpIpClient.cs

```diff
@@ -0,0 +1,1048 @@
+                    {
+                        Debug.Console(1, this, "Connect attempt failed {0}", o.ClientStatus);
+                        CheckClosedAndTryReconnect();
+                    }
+                });
+            }
+            catch (Exception ex)
+            {
+                Debug.Console(0, this, Debug.ErrorLogLevel.Error, "Client connection exception: {0}", ex.Message);
+                IsTryingToConnect = false;
+                CheckClosedAndTryReconnect();
+            }
+        }
+
+        /// <summary>
+        /// 
+        /// </summary>
+        public void Disconnect()
+        {
+            Debug.Console(2, "Disconnect Called");
+
+            DisconnectCalledByUser = true;
+
+            // stop trying reconnects, if we are
+            if (RetryTimer != null)
+            {
+                RetryTimer.Stop();
+                RetryTimer = null;
+            }
+
+            if (Client != null)
+            {
+                DisconnectClient();
+                Debug.Console(1, this, "Disconnected");
+            }
+        }
+
+        /// <summary>
+        /// Does the actual disconnect business
+        /// </summary>
+        public void DisconnectClient()
+        {
+            if (Client == null) return;
+
+            Debug.Console(1, this, "Disconnecting client");
+            if (IsConnected)
+                Client.DisconnectFromServer();
+
+            // close up client. ALWAYS use this when disconnecting.
+            IsTryingToConnect = false;
+
+            Debug.Console(2, this, "Disconnecting Client {0}", DisconnectCalledByUser ? ", Called by user" : "");
+            Client.SocketStatusChange -= Client_SocketStatusChange;
+            Client.Dispose();
+            Client = null;
+
+            if (ConnectFailTimer == null) return;
+            ConnectFailTimer.Stop();
+            ConnectFailTimer.Dispose();
+            ConnectFailTimer = null;
+        }
+
+        /// <summary>
+        ///  Internal call to close up client. ALWAYS use this when disconnecting.
+        /// </summary>
+        //void Cleanup()
+        //{
+        //    IsTryingToConnect = false;
+
+        //    if (Client != null)
+        //    {
+        //        //SecureClient.DisconnectFromServer();
+        //        Debug.Console(2, this, "Disconnecting Client {0}", DisconnectCalledByUser ? ", Called by user" : "");
+        //        Client.SocketStatusChange -= Client_SocketStatusChange;
+        //        Client.Dispose();
+        //        Client = null;
+        //    }
+        //    if (ConnectFailTimer != null)
+        //    {
+        //        ConnectFailTimer.Stop();
+        //        ConnectFailTimer.Dispose();
+        //        ConnectFailTimer = null;
+        //    }
+        //}
+
+
+
+
+
+
+
+
+
+
+
+
+
+        /// <summary>
+        /// Band aid delegate for choked server
+        /// </summary>
+        internal delegate void ConnectionHasHungCallbackDelegate();
```

Comment:
This can probably be removed. I'm guessing it was put in the original class to allow for handling Crestron's SecureTcpServer connections

Reply:
Removed unused delegate.